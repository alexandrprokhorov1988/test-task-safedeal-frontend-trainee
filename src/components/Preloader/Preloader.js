import React from 'react';
import './Preloader.css';

function Preloader() {

  return (
    <div className="preloader">
      <div className="preloader__img"/>
      <p className="preloader__text">Идет поиск фотографий...</p>
    </div>
  );
}

export default Preloader;
