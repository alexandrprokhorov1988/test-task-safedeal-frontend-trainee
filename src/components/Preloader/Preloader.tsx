import React from 'react';
import './Preloader.css';

/**
 * Preloader component.
 */
const Preloader = () => {

  return (
    <div className="preloader">
      <div className="preloader__img"/>
      <p className="preloader__text">Идет поиск фотографий...</p>
    </div>
  );
};

export default Preloader;
