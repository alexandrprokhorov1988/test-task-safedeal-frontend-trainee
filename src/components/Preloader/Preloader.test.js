import React from 'react';
import {unmountComponentAtNode} from "react-dom";
import renderer from 'react-test-renderer';
import Preloader from './Preloader';

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('Preloader component', () => {
  const render = renderer
    .create(<Preloader/>)
    .toJSON();
  expect(render).toMatchSnapshot();
});
