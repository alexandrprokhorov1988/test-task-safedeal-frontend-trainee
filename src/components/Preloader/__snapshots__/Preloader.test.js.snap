// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Рендер компонента Preloader 1`] = `
<div
  className="preloader"
>
  <div
    className="preloader__img"
  />
  <p
    className="preloader__text"
  >
    Идет поиск фотографий...
  </p>
</div>
`;
